apps/omg/lib/omg/fees/filter.ex

```diff
@@ -0,0 +1,140 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Fees.Filter do
+  @moduledoc """
+  Filtering of fees.
+  """
+
+  alias OMG.Fees
+
+  @doc ~S"""
+  Returns a filtered map of fees given a list of transaction types and currencies.
+  Passing a nil value or an empty array skip the filtering.
+
+  ## Examples
+
+      iex> OMG.Fees.Filter.filter(
+      ...>   %{
+      ...>     1 => %{
+      ...>       "eth" => %{
+      ...>         amount: 1,
+      ...>         subunit_to_unit: 1_000_000_000_000_000_000,
+      ...>         pegged_amount: 4,
+      ...>         pegged_currency: "USD",
+      ...>         pegged_subunit_to_unit: 100,
+      ...>         updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>       },
+      ...>       "omg" => %{
+      ...>         amount: 3,
+      ...>         subunit_to_unit: 1_000_000_000_000_000_000,
+      ...>         pegged_amount: 4,
+      ...>         pegged_currency: "USD",
+      ...>         pegged_subunit_to_unit: 100,
+      ...>         updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>       }
+      ...>     },
+      ...>     2 => %{
+      ...>       "omg" => %{
+      ...>         amount: 3,
+      ...>         subunit_to_unit: 1_000_000_000_000_000_000,
+      ...>         pegged_amount: 4,
+      ...>         pegged_currency: "USD",
+      ...>         pegged_subunit_to_unit: 100,
+      ...>         updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>       }
+      ...>     },
+      ...>     3 => %{
+      ...>       "omg" => %{
+      ...>         amount: 3,
+      ...>         subunit_to_unit: 1_000_000_000_000_000_000,
+      ...>         pegged_amount: 4,
+      ...>         pegged_currency: "USD",
+      ...>         pegged_subunit_to_unit: 100,
+      ...>         updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+      ...>       }
+      ...>     }
+      ...>   },
+      ...>   [1,2],
+      ...>   ["eth"]
+      ...> )
+      {:ok,
+        %{
+          1 => %{
+            "eth" => %{
+              amount: 1,
+              subunit_to_unit: 1_000_000_000_000_000_000,
+              pegged_amount: 4,
+              pegged_currency: "USD",
+              pegged_subunit_to_unit: 100,
+              updated_at: DateTime.from_iso8601("2019-01-01T10:10:00+00:00")
+            }
+          },
+          2 => %{}
+        }
+      }
+
+  """
+  @spec filter(Fees.full_fee_t(), list(pos_integer()), list(String.t()) | nil) ::
```

Comment:
are you missing `{:error, :tx_type_not_supported}`?

Reply:
based on the test you are: https://github.com/omisego/elixir-omg/pull/1219/files#diff-c8cf25565e2fe63edde08a01b11e386aR112

it's a bit worrying that dialyzer didn't catch this. might be worth restructuring (try simplifying) the module.